CMakeLists.txt

```diff
@@ -316,7 +300,8 @@ add_subdirectory(include)
 if(NOT MSVC)
 	add_subdirectory(man)
 endif()
-if(NOT MSVC OR ENABLE_VSTEST)
+# Tests require the openssl executable and are unavailable when building shared libraries
+if(LIBRESSL_APPS AND LIBRESSL_TESTS AND NOT BUILD_SHARED_LIBS)
```

Comment:
It appears that in the `.sym` files that certain functions are internal and not exposed in the .so. Because of this tests are disabled on `BUILD_SHARED_LIBS=ON`. I think this is the root cause of why the tests were failing on OpenBSD.